src/functions.js

```diff
@@ -0,0 +1,138 @@
+// select variables
+const todoListElement = document.getElementById('todo');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+const todoInput = document.querySelector('input');
+let EditTodoId = -1;
+
+// save Todo function to the local storage of a user machine
+
+export function saveTodo() {
+  const todoValue = todoInput.value;
+  const isEmpty = todoValue === '';
+  // check duplicate input
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+  // check if userenters an  empty input and clicks
+  if (isEmpty) {
+    alert('input is empty');
+  } else if (isDuplicate) {
+    alert('already exists');
+  } else {
+    if (EditTodoId >= 0) {
+      todos = todos.map((todo, index) => ({
+        ...todo,
+
+        value: index === EditTodoId ? todoValue : todo.value,
+      }));
+      EditTodoId = -1;
+    } else {
+      todos.push({
+        value: todoValue,
+        completed: false,
+        index: todos.length + 1,
+      });
+    }
+    todoInput.value = '';
+  }
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
+// render to user interphase
+export function renderTodos() {
```

Comment:
use arrow functions instead of the function keyword
const renderTodos = () => {
if (todos.length === 0) {